src/system/mod.rs

```diff
@@ -35,19 +36,58 @@ pub mod term;
 
 pub mod wait;
 
-/// Mark all the file descriptors that are not the IO streams to be closed when `exec` is called.
-pub(crate) fn close_the_universe() -> io::Result<()> {
-    let min_fd = libc::STDERR_FILENO + 1;
-    let max_fd: c_int = -1;
-
-    cerr(unsafe {
-        libc::syscall(
-            libc::SYS_close_range,
-            min_fd,
-            max_fd,
-            libc::CLOSE_RANGE_CLOEXEC,
-        )
-    })?;
+/// A type able to close every file descriptor except for the ones pased via [`FileCloser::except`]
+/// and the IO streams.
+pub(crate) struct FileCloser {
+    fds: BTreeSet<c_uint>,
+}
+
+impl FileCloser {
+    pub(crate) const fn new() -> Self {
+        Self {
+            fds: BTreeSet::new(),
+        }
+    }
+
+    pub(crate) fn except<F: AsRawFd>(&mut self, fd: &F) {
+        self.fds.insert(fd.as_raw_fd() as c_uint);
+    }
+
+    /// Close every file descriptor that is not one of the IO streams or one of the file
+    /// descriptors passed via [`FileCloser::except`].
+    pub(crate) fn close_the_universe(self) -> io::Result<()> {
+        let mut fds = self.fds.into_iter();
+
+        let Some(mut curr_fd) = fds.next() else {
+            return close_range(STDERR_FILENO as c_uint + 1, c_uint::MAX);
```

Comment:
I think this first `close_range` needs more checks because you can do `except(0)`. or you could make `except` panic if it receives a fd smaller than 3.

Reply:
I added a test calling `except` over `stdin/out/err` and it works. The only thing this will do is that we will try to call
```rust
close_range(3, 0);  // curr_fd = 0
close_range(1, 0);  // curr_fd = 0, next_fd = 1
close_range(2, 1);  // curr_fd = 1, next_fd = 2
close_range(2, -1); // curr_fd = 2
```
and the first three will just return `Ok(())` instead of actually doing the syscall.

Reply:
I added a test calling `except` over `stdin/out/err` and it works. The only thing this will do is that we will try to call
```rust
close_range(1, 0);  // curr_fd = 0, next_fd = 1
close_range(2, 1);  // curr_fd = 1, next_fd = 2
close_range(3, -1); // curr_fd = 2
```
and the first two will just return `Ok(())` instead of actually doing the syscall.